R/twas.R

```diff
@@ -464,11 +464,23 @@ twas_weights <- function(X, Y, weight_methods, num_threads = 1, seed = NULL) {
     # Hardcoded vector of multivariate methods
     multivariate_weight_methods <- c("mrmash_weights", "mvsusie_weights")
     args <- weight_methods[[method_name]]
-
+    
     # Remove columns with zero standard error
     valid_columns <- apply(X, 2, function(col) sd(col) != 0)
-    X_filtered <- X[, valid_columns]
-
+    X_filtered <- as.matrix(X[, valid_columns])
+                           
+    if (method_name=="susie_weights"){
```

Comment:
This is good thinking but I don't think we should handle it for users here. We should throw an error message. That is, in the `susie_weight` function if the number of X columns does not match the dimension of the fitted model, we throw an error. Woudl you implement that instead and remove your change here?

Reply:
@gaow This has been addressed